sdks/java/core/src/test/java/org/apache/beam/sdk/runners/dataflow/TestCountingSource.java

```diff
@@ -50,6 +50,8 @@
 
   private static List<Integer> finalizeTracker;
   private final int numMessagesPerShard;
+  // If non-negative, this is the number of splits returned by generateInitialSplits.
+  private final int numSplits;
```

Comment:
no point in having the allowSplitting attribute as well, numSplits subsumes its responsibilities, please remove allowSplitting boolean and just set numSplits to 1 when a person call withoutSplits

Please also add a test to https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/runners/dataflow/TestCountingSourceTest.java to ensure that withoutSplitting and setting numSplits does what it asks